src/oid4vp/holder.rs

```diff
@@ -0,0 +1,511 @@
+}
+
+#[uniffi::export(async_runtime = "tokio")]
+impl Holder {
+    /// Uses VDC collection to retrieve the credentials for a given presentation definition.
+    #[uniffi::constructor]
+    pub async fn new(
+        vdc_collection: Arc<VdcCollection>,
+        // request_signer: Arc<dyn RequestSignerInterface>,
+        trusted_dids: Vec<String>,
+    ) -> Result<Arc<Self>, OID4VPError> {
+        let client = oid4vp::core::util::ReqwestClient::new()
+            .map_err(|e| OID4VPError::HttpClientInitialization(e.to_string()))?;
+
+        Ok(Arc::new(Self {
+            client,
+            vdc_collection: Some(vdc_collection),
+            metadata: WalletMetadata::openid4vp_scheme_static(),
+            trusted_dids,
+            // request_signer: Some(request_signer),
+            provided_credentials: None,
+            authorization_request: RwLock::new(None),
+        }))
+    }
+
+    /// Construct a new holder with provided credentials
+    /// instead of a VDC collection.
+    ///
+    /// This constructor will use the provided credentials for the presentation,
+    /// instead of searching for credentials in the VDC collection.
+    #[uniffi::constructor]
+    pub async fn new_with_credentials(
+        provided_credentials: Vec<Arc<Credential>>,
+        // _request_signer: Arc<dyn RequestSignerInterface>,
+        trusted_dids: Vec<String>,
+    ) -> Result<Arc<Self>, OID4VPError> {
+        let client = oid4vp::core::util::ReqwestClient::new()
+            .map_err(|e| OID4VPError::HttpClientInitialization(e.to_string()))?;
+
+        Ok(Arc::new(Self {
+            client,
+            vdc_collection: None,
+            metadata: WalletMetadata::openid4vp_scheme_static(),
+            trusted_dids,
+            // request_signer: Some(request_signer),
+            provided_credentials: Some(provided_credentials),
+            authorization_request: RwLock::new(None),
+        }))
+    }
+
+    /// Given an authorization request URL, return a permission request,
+    /// which provides a list of requested credentials and requested fields
+    /// that align with the presentation definition of the request.
+    ///
+    /// This will fetch the presentation definition from the verifier.
+    pub async fn authorization_request(
+        &self,
+        url: Url,
+    ) -> Result<Arc<PermissionRequest>, OID4VPError> {
+        let request = self
+            .validate_request(url)
+            .await
+            .map_err(|e| OID4VPError::RequestValidation(format!("{e:?}")))?;
+
+        // Store the authorization request object for later use.
+        self.authorization_request
+            .write()
+            .await
+            .replace(request.clone());
+
+        match request.response_mode() {
+            ResponseMode::DirectPost | ResponseMode::DirectPostJwt => {
+                self.permission_request(&request).await
+            }
+            // TODO: Implement support for other response modes.
```

Comment:
Can you remove this comment from the code and just ensure we have a ticket/issue for this?

Reply:
I removed the comment and have opened a ticket for tracking other response modes.